collection.go

```diff
@@ -117,3 +117,17 @@ func (c Collection[V]) Last() V {
 
 	return c.values[c.keys[len(c.keys)-1]]
 }
+
+func (c Collection[V]) Merge(other Collection[V]) (Collection[V], error) {
+	newCollection := makeCollection[V](0)
+
+	c.Each(func(_ any, v V) {
+		newCollection.Push(v)
```

Comment:
By the way, definitely use `Put` here, as you need to preserve the collection keys.